internal/util/vault.go

```diff
@@ -89,14 +96,38 @@ func InitVaultKMS(kmsID string, config, secrets map[string]string) (EncryptionKM
 	if !ok || kms.VaultAddress == "" {
 		return nil, fmt.Errorf("missing 'vaultAddress' for vault KMS %s", kmsID)
 	}
-	kms.VaultAuthPath, ok = config["vaultAuthPath"]
-	if !ok || kms.VaultAuthPath == "" {
-		kms.VaultAuthPath = vaultDefaultAuthPath
+	kms.VaultTokenAuth = false
+	authToken, ok := config["vaultTokenAuth"]
+	if ok {
+		kms.VaultTokenAuth, err = strconv.ParseBool(authToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse 'vaultTokenAuth' for vault <%s> kms config: %s",
+				kmsID, err)
+		}
 	}
-	kms.VaultRole, ok = config["vaultRole"]
-	if !ok || kms.VaultRole == "" {
-		kms.VaultRole = vaultDefaultRole
+	// Check if vaultTokenAuth is enabled and if yes, use token based auth
+	if kms.VaultTokenAuth {
+		// Parse VaultToken from secret
+		vaultTokenFromSecret, ok = config["vaultTokenFromSecret"]
+		if ok && vaultTokenFromSecret != "" {
```

Comment:
what if the token is empty in secret?

Reply:
sure, we could return `err` in that case :+1:

Reply:
Done.